### PR TITLE
fix(planner:create_button_validation): Add more validation to create iteration form

### PR DESCRIPTION
**Pull request by karthikjeeyar**
This PR contain following fixes
1. Disables the create/update iteration form if the iteration name field is empty.
2. Code formatting as per new IDE settings.

Fixes: https://github.com/openshiftio/openshift.io/issues/2650

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @karthikjeeyar for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

File: packages/planner/src/app/components_ngrx/iterations-modal/iterations-modal.component.html [hunk 1]
```diff
@@ -28,76 +17,59 @@
...
-        id="cancel-iteration-button" type="submit"
+      <button [disabled]="submitLoading" class="btn btn-default margin-right-10" id="cancel-iteration-button" type="submit"
         value="0" (click)="createUpdateIterationDialog.close()">
         Cancel
       </button>
-      <button class="btn btn-primary"
-        [disabled]="submitLoading"
-        id="create-iteration-button" type="submit" value="0"
-        (click)="createUpdateIterationDialog.onSubmit.emit(undefined)">
+      <button class="btn btn-primary" [disabled]="isNameFieldEmpty ||submitLoading" id="create-iteration-button" type="submit"
```
Comment (divyanshiGupta) [inline packages/planner/src/app/components_ngrx/iterations-modal/iterations-modal.component.html]: @karthikjeeyar there should be a space between `||` and `submitLoading` and anyways the html formatting looks different from the way it should be.

**Comment by divyanshiGupta** [comment]
Please resolve conflicts

**Comment by karthikjeeyar** [comment]
@christianvogt  i have formatted the code again via prettier.

**Comment by joshuawilson** [comment]
Please rebase.

File: packages/planner/src/app/components_ngrx/iterations-modal/iterations-modal.component.ts [hunk 2]
```diff
@@ -163,12 +163,16 @@ export class FabPlannerIterationModalComponent implements OnInit, OnDestroy, OnC
...
 
-  ngOnChanges() {}
+  ngOnChanges() { }
 
   ngOnDestroy() {
     // prevent memory leak when component is destroyed
   }
 
+  get isNameFieldEmpty(): boolean {
+    return this.iteration.name.trim().length <= 0;
```
Comment (joshuawilson) [inline packages/planner/src/app/components_ngrx/iterations-modal/iterations-modal.component.ts]: Is there a reason you didn't use `<==` instead?

Comment (christianvogt) [reply packages/planner/src/app/components_ngrx/iterations-modal/iterations-modal.component.ts]: It does not exist :P

Comment (joshuawilson) [reply packages/planner/src/app/components_ngrx/iterations-modal/iterations-modal.component.ts]: :face_palm:

**Pull request merged**